go/arrow/avro/schema.go

```diff
@@ -0,0 +1,512 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package avro reads Avro OCF files and presents the extracted data as records
+
+package avro
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"strconv"
+
+	"github.com/apache/arrow/go/v13/arrow"
+	"github.com/apache/arrow/go/v13/internal/types"
+)
+
+type schemaNode struct {
+	name        string
+	ofType      interface{}
+	logicalType string
+	precision   int32
+	scale       int32
+	size        int
+	symbols     []string
+	fields      []interface{}
+}
+
+// ArrowSchemaFromAvro returns a new Arrow schema from an Avro schema JSON.
+func ArrowSchemaFromAvro(avroSchema []byte) (*arrow.Schema, error) {
+	var m map[string]interface{}
+	var node schemaNode
+	json.Unmarshal(avroSchema, &m)
+	if name, ok := m["name"]; ok {
+		node.name = name.(string)
+	} else {
+		node.name = "NameStub"
+	}
```

Comment:
were we gonna try using the hambra library for this?

Reply:
The input schema is a json in canonical format which is identical whether it's coming from hamba/avro or linkedin/goavro. 
With the changes suggested previously the current code produces a valid arrow schema and only needs to be run once per avro OCF so I focused my efforts on the reader which I did move over to use /hamba/avro.  
Moving the schema converter to hamba might be better for a future version if we want to add extended support for Avro unions beyond the currently supported "null + 1 on other type".